src/mito2/src/manifest/manager.rs

```diff
@@ -14,20 +14,32 @@
 
 use std::sync::Arc;
 
-use common_telemetry::{debug, info};
-use store_api::manifest::action::{ProtocolAction, ProtocolVersion};
+use common_datasource::compression::CompressionType;
+use common_telemetry::{debug, info, warn};
+use object_store::ObjectStore;
 use store_api::manifest::{ManifestVersion, MAX_VERSION, MIN_VERSION};
 use tokio::sync::RwLock;
 
 use crate::error::Result;
 use crate::manifest::action::{
     RegionChange, RegionCheckpoint, RegionManifest, RegionManifestBuilder, RegionMetaAction,
-    RegionMetaActionIter, RegionMetaActionList,
+    RegionMetaActionList,
 };
-use crate::manifest::options::RegionManifestOptions;
 use crate::manifest::storage::ManifestObjectStore;
 use crate::metadata::RegionMetadataRef;
 
+/// Options for [RegionManifestManager].
+#[derive(Debug, Clone)]
+pub struct RegionManifestOptions {
+    /// Directory to store manifest.
+    pub manifest_dir: String,
+    pub object_store: ObjectStore,
+    pub compress_type: CompressionType,
+    /// Interval of version ([ManifestVersion](store_api::manifest::ManifestVersion)) between two checkpoints.
+    /// Set to 0 to disable checkpoint.
+    pub checkpoint_interval: u64,
```

Comment:
I think `margin` is better. Most of time `interval` represents a time interval.

Reply:
it looks like `interval` is more general, from my search https://nightlies.apache.org/flink/flink-docs-master/docs/ops/state/large_state_tuning/#tuning-checkpointing